src/puppetlabs/puppetdb/queue.clj

```diff
@@ -103,6 +106,16 @@
       (utf8-truncate sanitized-certname trunc-length)
       sanitized-certname)))
 
+(defn received+producer-ts [received-ts producer-ts]
```

Comment:
Would you mind adding a docstring to this function? I found this function a little mysterious the first time I saw it :)